firestore/src/common/document_change.cc

```diff
@@ -104,5 +104,10 @@ std::size_t DocumentChange::new_index() const {
   return internal_->new_index();
 }
 
+bool operator==(const DocumentChange& lhs, const DocumentChange& rhs) {
+  return lhs.type() == rhs.type() && lhs.old_index() == rhs.old_index() &&
```

Comment:
Ideally, we should delegate this to the `internal_` object. Can you add a TODO to that effect and make the corresponding change in the iOS repo? Then the TODO could be fixed once the iOS change gets released.

Reply:
I had looked at document_change_main.h and since it didn't have `operator==`, I thought that the iOS sdk doesn't have it. I just double-checked and found that the iOS sdk does in fact [implement](https://github.com/firebase/firebase-ios-sdk/blob/bdac2b87f6d6e604deae914a970a75136b3a62b5/Firestore/core/src/api/document_change.h#L80) this. I'll rewrite this to delegate to the underlying implementations.